software/controller/test/commfail_alarm/commfail_alarm_test.cpp

```diff
@@ -0,0 +1,68 @@
+/* Copyright 2020, RespiraWorks
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "commfail_alarm.h"
+#include "gtest/gtest.h"
+#include <cmath>
+#include <tuple>
+
+// To test Handler function
+TEST(CommFailAlarmTest, AlarmRaiseHandler) {
+  Time ts = microsSinceStartup(0);               // start
+  constexpr Time lastrx = microsSinceStartup(0); // start
+  constexpr Duration step = milliseconds(10);
+  CommFailAlarm cfa;
+
+  cfa.Initialize(ts);
+  for (int i = 0; i < 20; i++) {
+    ts += step;
+    if (ts - lastrx < milliseconds(101)) {
```

Comment:
So you are only testing the correctness of line 39 in `commfail_alarm.cpp` and you are testing it with an identical comparison operation. This does not seem like a strong test.

Reply:
added test cases to cover branches & variables of the code